### ISSUE TITLE
Shell should stop accepting new commands when it sees quit or reaches end of input stream

### ISSUE THREAD

**Issue by edgariscoding** [unt-sysprog-team/major-assignment-1#3]
Your shell can be run in two ways: (1) interactive and (2) batch. In interactive mode, you will display a prompt (any string of your choosing) and the user of the shell will type in a command at the prompt. In batch mode, your shell is started by specifying a batch file on its command line. The batch file contains the list of commands that should be executed. In batch mode, you should not display a prompt. In batch mode, you should echo each line you read from the batch file back to the user before executing it. This will help you when you debug your shells. **In both interactive and batch mode, your shell stops accepting new commands when it sees the quit command on a line or reaches the end of the input stream (i.e., the end of the batch file or the user types ‘Ctrl-D’). The shell should then exit after all running processes have terminated.**

**To exit the shell, the user can type quit. This should just exit the shell and be done with it (the exit() system call will be useful here). Note that quit is a built-in shell command. It is not to be executed like other programs the user types in. If the quit command is on the same line with other commands, you should ensure that the other commands execute (and finish) before you exit your shell.**

These are all valid examples for quitting the shell:

```
prompt> quit
prompt> quit; cat file1
prompt> cat file1; quit
```

**Comment by edgariscoding** [comment]
I assume that this means that if this is input:

```
prompt> cat file1; quit
```

Then `cat file1` will be executed and then the shell will `quit`.

But if this is input:

```
prompt> quit; cat file1
```

Then the shell will `quit` and `cat file1` will not execute.

May be worth getting clarification on this though.

**Comment by edgariscoding** [comment]
**Implemented in #18.**

If Shell is given the `quit` command either in interactive mode or within a batch file, the program will exit. 

If the command is given in the middle of a string of commands, only the commands leading up to the `quit` command will be executed.

![screenshot](http://i.imgur.com/S6WbSLe.png)
